influxql/iterator.go

```diff
@@ -539,3 +585,102 @@ type nilFloatIterator struct{}
 
 func (*nilFloatIterator) Close() error      { return nil }
 func (*nilFloatIterator) Next() *FloatPoint { return nil }
+
+// integerReduceSliceFloatIterator executes a reducer on all points in a window and buffers the result.
+// This iterator receives an integer iterator but produces a float iterator.
+type integerReduceSliceFloatIterator struct {
```

Comment:
We do need cast iterators but for reductions it's going to cause a _lot_ of overhead to copy every point.